### DISCUSSION TITLE
Oracle Linux Server 8.5

### DISCUSSION THREAD

**Discussion by orlakwahr** [autobase-tech/autobase#129] [General]
Hello.

when target os is Oracle Linux Server 8.5
it tries to install os_specific_packages for RedHat 8 and basically all that is for RedHat 8 get's applied.
while there are no packages in Oracle like python3-devel or  python2, python3-libselinux, python3-libsemanage
I think some other things that might get applied from RehHat 8 might not work out for Oracle Linux Server 8.5

failed: [XXX.XXX.XXX.XXX] (item=postgresql13-devel) => {"changed": false, "failures": [], "item": "postgresql13-devel", "msg": "Depsolve Error occured: \n Problem: cannot install the best candidate for the job\n  - nothing provides clang-devel >= 8.0.1 needed by postgresql13-devel-13.5-1PGDG.rhel8.x86_64\n  - nothing provides llvm-devel >= 8.0.1 needed by postgresql13-devel-13.5-1PGDG.rhel8.x86_64\n  - nothing provides libicu-devel needed by postgresql13-devel-13.5-1PGDG.rhel8.x86_64", "rc": 1, "results": []}

what can be done to install make this work with OL8?

thanks!

**Comment by vitabaks** [comment]
Hi @orlakwahr 

This is very strange since daily tests show no problems with the availability of packages. \
https://github.com/vitabaks/postgresql_cluster/actions/workflows/schedule_pg_oracle_linux8.yml


> while there are no packages in Oracle like python3-devel or python2, python3-libselinux, python3-libsemanage

these packages are present: 

<img width="927" alt="image" src="https://user-images.githubusercontent.com/37010174/143790303-8f8ca772-abf8-4065-9c99-a6a117354ae8.png">

`os_specific_packages.RedHat-8` packages for the red hat family (including oracle) are listed here, not just for the Red Hat OS.

**Comment by vitabaks** [comment]
> nothing provides clang-devel 
> nothing provides llvm-devel 
> nothing provides libicu-devel 


I'll check and give you an answer a little later.

**Comment by orlakwahr** [comment]
well I have just used the official image from oracle website. it does not work out of the box. should we compare the repos? As I see it - anything that has -devel in name is not there.

**Comment by orlakwahr** [comment]
ok there is python36-devel in ol8_appstream

**Comment by orlakwahr** [comment]
but if I enable it I get  FAILED! => {"changed": false, "msg": "Failed to download metadata for repo 'ol8_baseos_latest': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried", "rc": 1, "results": []}
and
 FAILED! => {"changed": false, "msg": "Failed to download metadata for repo 'ol8_appstream': Cannot download repomd.xml: Cannot download repodata/repomd.xml: All mirrors were tried", "rc": 1, "results": []}

**Comment by orlakwahr** [comment]
it does work as intended using a fresh debian install for the host.

**Comment by orlakwahr** [comment]
patroni will not build if we do not install python36-devel


FAILED! => {"changed": false, "cmd": ["/usr/bin/pip3", "install", "-U", "--trusted-host=pypi.python.org", "--trusted-host=pypi.org", "--trusted-host=files.pythonhosted.org", "patroni"], "msg": "stdout: Collecting patroni\n  Downloading https://files.pythonhosted.org/packages/7d/6b/7f758dcbe619ca5688fdd36d7e1c0cc36900f6dd2a4f4491ec438c52cd97/patroni-2.1.1-py3-none-any.whl (219kB)\nCollecting six>=1.7 (from patroni)\n  Downloading https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl\nRequirement already up-to-date: prettytable>=0.7 in /usr/local/lib/python3.6/site-packages (from patroni)\nCollecting psutil>=2.0.0 (from patroni)\n  Downloading https://files.pythonhosted.org/packages/e1/b0/7276de53321c12981717490516b7e612364f2cb372ee8901bd4a66a000d7/psutil-5.8.0.tar.gz (470kB)\nCollecting python-dateutil (from patroni)\n  Downloading https://files.pythonhosted.org/packages/36/7a/87837f39d0296e723bb9b62bbb257d0355c7f6128853c78955f57342a56d/python_dateutil-2.8.2-py2.py3-none-any.whl (247kB)\nRequirement already up-to-date: ydiff>=1.2.0 in /usr/local/lib/python3.6/site-packages (from patroni)\nRequirement already up-to-date: urllib3!=1.21,>=1.19.1 in /usr/local/lib/python3.6/site-packages (from patroni)\nCollecting PyYAML (from patroni)\n  Downloading https://files.pythonhosted.org/packages/b3/85/79b9e5b4e8d3c0ac657f4e8617713cca8408f6cdc65d2ee6554217cedff1/PyYAML-6.0-cp36-cp36m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (603kB)\nRequirement already up-to-date: click>=4.1 in /usr/local/lib/python3.6/site-packages (from patroni)\nRequirement already up-to-date: wcwidth in /usr/local/lib/python3.6/site-packages (from prettytable>=0.7->patroni)\nRequirement already up-to-date: importlib-metadata; python_version < \"3.8\" in /usr/local/lib/python3.6/site-packages (from prettytable>=0.7->patroni)\nRequirement already up-to-date: zipp>=0.5 in /usr/local/lib/python3.6/site-packages (from importlib-metadata; python_version < \"3.8\"->prettytable>=0.7->patroni)\nRequirement already up-to-date: typing-extensions>=3.6.4; python_version < \"3.8\" in /usr/local/lib/python3.6/site-packages (from importlib-metadata; python_version < \"3.8\"->prettytable>=0.7->patroni)\nInstalling collected packages: six, psutil, python-dateutil, PyYAML, patroni\n  Running setup.py install for psutil: started\n    Running setup.py install for psutil: finished with status 'error'\n    Complete output from command /usr/bin/python3.6 -u -c \"import setuptools, tokenize;__file__='/tmp/pip-build-d5n5hp30/psutil/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\\r\\n', '\\n');f.close();exec(compile(code, __file__, 'exec'))\" install --record /tmp/pip-22biy5_x-record/install-record.txt --single-version-externally-managed --compile:\n    running install\n    /usr/local/lib/python3.6/site-packages/setuptools/command/install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.\n      setuptools.SetuptoolsDeprecationWarning,\n    running build\n    running build_py\n    creating build\n    creating build/lib.linux-x86_64-3.6\n    creating build/lib.linux-x86_64-3.6/psutil\n    copying psutil/_pswindows.py -> build/lib.linux-x86_64-3.6/psutil\n    copying psutil/_psbsd.py -> build/lib.linux-x86_64-3.6/psutil\n    copying psutil/_compat.py -> build/lib.linux-x86_64-3.6/psutil\n    copying psutil/_common.py -> build/lib.linux-x86_64-3.6/psutil\n    copying psutil/_psposix.py -> build/lib.linux-x86_64-3.6/psutil\n    copying psutil/__init__.py -> build/lib.linux-x86_64-3.6/psutil\n    copying psutil/_psaix.py -> build/lib.linux-x86_64-3.6/psutil\n    copying psutil/_pslinux.py -> build/lib.linux-x86_64-3.6/psutil\n    copying psutil/_pssunos.py -> build/lib.linux-x86_64-3.6/psutil\n    copying psutil/_psosx.py -> build/lib.linux-x86_64-3.6/psutil\n    creating build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_aix.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_system.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_testutils.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_bsd.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_connections.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_memleaks.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_unicode.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_posix.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_windows.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_misc.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_sunos.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/__init__.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_osx.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_process.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_linux.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/__main__.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/test_contracts.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    copying psutil/tests/runner.py -> build/lib.linux-x86_64-3.6/psutil/tests\n    running build_ext\n    building 'psutil._psutil_linux' extension\n    creating build/temp.linux-x86_64-3.6\n    creating build/temp.linux-x86_64-3.6/psutil\n    gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=580 -DPSUTIL_LINUX=1 -I/usr/include/python3.6m -c psutil/_psutil_common.c -o build/temp.linux-x86_64-3.6/psutil/_psutil_common.o\n    psutil/_psutil_common.c:9:10: fatal error: Python.h: No such file or directory\n     #include <Python.h>\n              ^~~~~~~~~~\n    compilation terminated.\n    error: command 'gcc' failed with exit status 1\n    \n    ----------------------------------------\n\n:stderr: WARNING: Running pip install with root privileges is generally not a good idea. Try `pip3 install --user` instead.\n/usr/local/lib/python3.6/site-packages/setuptools/command/install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.\n  setuptools.SetuptoolsDeprecationWarning,\nCommand \"/usr/bin/python3.6 -u -c \"import setuptools, tokenize;__file__='/tmp/pip-build-d5n5hp30/psutil/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\\r\\n', '\\n');f.close();exec(compile(code, __file__, 'exec'))\" install --record /tmp/pip-22biy5_x-record/install-record.txt --single-version-externally-managed --compile\" failed with error code 1 in /tmp/pip-build-d5n5hp30/psutil/\n"}
